com.sap.cloud.lm.sl.cf.core/src/main/java/com/sap/cloud/lm/sl/cf/core/exec/MethodExecution.java

```diff
@@ -0,0 +1,38 @@
+package com.sap.cloud.lm.sl.cf.core.exec;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+
+import com.sap.cloud.lm.sl.cf.core.exec.MethodExecution.ExecutionState;
+
+public class MethodExecution<T> {
```

Comment:
Isn't that HttpMethodExecution?

Reply:
The underlying method behaviour is hidden from us and we should not consider it. What matters is that it can be either finished or still executing.